kubeflow/testing
# Support subdirectories for junit files and junit_class_name

Support subdirectories for junit files and grouping in test grid by name
  
  * We want to support putting junit files in subdirectories e.g.
    junit/junit_workflow1/junit_test1.xml
    junit/junit_workflow2/junit_test1.xml
  
    * prow_artifacts.py needs to fetch the junit files from GCS in order to
      determine if there were any test failures to properly set the prow job
      status
  
    * The code however was making an assumption that all results were in
      the artifacts dir and not subdirectories
  
  * run_e2e_workflow.py should set a keyword argument test_target_name
    to the name of the workflow (as set in prow_config.yaml) for each of
    the python functions creating argo workflows.
  
    * We want to use the junit attribute classname to group tests coming
      from the same workflow. To do that we need to know the name of the workflow.
  
    * We also use this as a prefix for the test name to support grouping
      hierarchically
  
  * Related to kubeflow/testing#489
  
  * Update kfunittests to Use a junit subdirectory to make sure subdirectories are properly handled.
  
  * Update docs
  
  * update toc.
  
  * Convert test_py_lint to use pytest. Do this because we want to be able to test the use of junit features.
  
  * Add a separate mkdir step.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/testing/490)
